docs: correct AWS `ProviderConfig` example

chr0n1x opened the pull request:
https://github.com/hashicorp/terraform-provider-aws/issues/29141#issuecomment-1414199622

<!--
Thank you for helping to improve Official Terraform Provider!

Please read through https://git.io/fj2m9 if this is your first time opening a
Official Terraform Provider pull request. Find us in https://slack.crossplane.io/messages/dev if
you need any help contributing.
-->

### Description of your changes

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open Official Terraform Provider issue. If yours does, you
can uncomment the below line to indicate which issue your PR fixes, for example
"Fixes #500":

-->

Addresses https://github.com/upbound/provider-terraform/issues/63

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change.
-->
https://github.com/hashicorp/terraform-provider-aws/issues/29141#issuecomment-1414199622

Upbound-CLA commented:
[![CLA assistant check](https://cla.upbound.io/pull/badge/not_signed)](https://cla.upbound.io/upbound/provider-terraform?pullRequest=71) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.upbound.io/upbound/provider-terraform?pullRequest=71) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.upbound.io/check/upbound/provider-terraform?pullRequest=71) it.</sub>

chr0n1x commented:
```
Running [/home/runner/golangci-lint-1.50.0-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  level=warning msg="[runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner. "
  level=warning msg="[runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive."
  level=warning msg="[runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref."
  level=warning msg="[runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused."
  level=warning msg="[runner] The linter 'varcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused."
  level=warning msg="[runner] The linter 'structcheck' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused."
  level=warning msg="[linters_context] rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/[26](https://github.com/upbound/provider-terraform/actions/runs/4087961198/jobs/7049613735#step:8:28)49."
  level=warning msg="[linters_context] sqlclosecheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
  level=warning msg="[linters_context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
  level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
```

????????

bobh66 reviewed (approved):
LGTM - thanks for following up on this!
